xonsh/tools.py

```diff
@@ -1585,3 +1585,12 @@ def _iglobpath(s, ignore_case=False, sort_result=None):
 def iglobpath(s, ignore_case=False, sort_result=None):
     """Simple wrapper around iglob that also expands home and env vars."""
     return _iglobpath(s, ignore_case=ignore_case, sort_result=sort_result)[0]
+
+
+def ensure_timestamp(t):
+    try:
```

Comment:
yes this the bare minimum to just pass the two tests. btw why do you prefer type checking over EAFP for this?